Automattic/wp-calypso
# update: add variants data and product selection

#### Proposed Changes

This PR adds logic to the Lightbox in regards to selecting Product variants when possible and displaying current pricing. 

#### Testing Instructions

* Run `git fetch && git checkout update/jetpack-product-lightbox-product-data`
  * Run `yarn start-jetpack-cloud`
  * Go to http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1

**OR**

* use the jetpack cloud live link and goto `/pricing?flags=jetpack/pricing-page-rework-v1

* Open lightbox by clicking `More about ...` button on any product card
* Verify that you see the Variant options on the right bar when applicable and that they are selectable.
* The price should be displayed correctly


https://user-images.githubusercontent.com/60262784/189656668-97991413-63aa-4335-9c10-bded198a6cdb.mov





*

#### Pre-merge Checklist






**Translation screenshots**
<img width="402" alt="Screenshot 2022-09-14 at 21 13 15" src="https://user-images.githubusercontent.com/60262784/190231192-53eb8d04-4930-4278-a992-8fa3eaf75764.png">

<img width="519" alt="Screenshot 2022-09-14 at 21 12 38" src="https://user-images.githubusercontent.com/60262784/190231162-433278a0-2ac2-44ec-beb9-53f6a3e8cc20.png">

<img width="494" alt="Screenshot 2022-09-14 at 21 13 44" src="https://user-images.githubusercontent.com/60262784/190231213-ad3b2428-a074-43a6-8223-868fd6b104cf.png">

<img width="144" alt="Screenshot 2022-09-14 at 21 14 59" src="https://user-images.githubusercontent.com/60262784/190231301-d61f40e7-ba8b-4fbd-bc92-6880a96cae2a.png">

Related to #
1202796695664022-as-1202796695664087/f

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202796695664087
  - 0-as-1202796695664090
